seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/SourceSplitEnumeratorTask.java

```diff
@@ -133,7 +135,7 @@ public ProgressState call() throws Exception {
     public void triggerBarrier(Barrier barrier) throws Exception {
         log.debug("split enumer trigger barrier [{}]", barrier);
         if (barrier.prepareClose()) {
-            this.currState = PREPARE_CLOSE;
```

Comment:
Note: The state machine state cannot be changed outside the state machine. Just should send event.